### PR TITLE
feat(icon): add displayName and __ui__ properties

### LINKED ISSUE #2870: `displayName` is not set for the component of "@yamada-ui/icon"
script: component-display-name
package: @yamada-ui/icon

### Description
The component "@yamada-ui/icon" uses `forwardRef`, but `displayName` is not set, making it difficult to see the display in [React Developer Tools](https://github.com/facebook/react/tree/main/packages/react-devtools).

### Additional Information
Set `displayName` and `__ui__` after each component definition.

For example, like this.

```tsx
const Component = forwardRef((props, ref) => {
  ...
})

Component.displayName = 'Component'
Component.__ui__ = 'Component'
```

Replace `'component'` with the name of the component. For example, for a `Select` component, it is `'Select'`.

`__ui__` is an identifier required for internal use by Yamada UI. Be sure to set it together with `displayName`.

### PR DESCRIPTION
**Pull request by KekmaTime**
Closes #2870 

## Description
 `displayName` is not set for the component of "@yamada-ui/icon"

## Current behavior (updates)
displayName is not set for the component as of now.

## New behavior

added displayName and __ui__ properties

### CODE REVIEW

File: .changeset/silly-starfishes-share.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/icon": patch
+---
+
+feat(icon): add `displayName` and `__ui` properties
```
Comment (taiseiiiii) [inline .changeset/silly-starfishes-share.md:5]: The correct term is `__ui__`, not `__ui`.

For unique terms, wrap the word with "`".
For example, "__ui__".

**Review by taiseiiiii — CHANGES_REQUESTED** [review]
Thanks for PR!!
Would it be possible to make a few modifications to this?

File: packages/components/icon/src/icon.tsx [hunk 2]
```diff
@@ -79,6 +79,9 @@ export const Icon = forwardRef<IconProps, "svg">((props, ref) => {
   )
 })
 
+Icon.displayName = "Icon"
+Icon.__ui__ = "Icon"
+
```
[the remark this answers is not in the document]
Comment (KekmaTime) [reply packages/components/icon/src/icon.tsx:84]: okay 👍🏽

**Comment by KekmaTime** [comment]
@taiseiiiii ive made all the changes you've requested

File: packages/components/icon/src/icon.tsx [hunk 3]
```diff
@@ -133,3 +136,9 @@ export const ChevronIcon: FC<IconProps> = (props) => {
     </Icon>
   )
 }
+
+CheckIcon.displayName = "CheckIcon"
```
Comment (taiseiiiii) [inline packages/components/icon/src/icon.tsx:140]: Please define `__ui__` for each component as well.

**Review by taiseiiiii — CHANGES_REQUESTED** [review]
@KekmaTime 
Thanks for your response!
Please make additional modifications.

File: packages/components/icon/src/icon.tsx (hunk 3 above)
Comment (KekmaTime) [reply packages/components/icon/src/icon.tsx:140]: its done.

**Comment by KekmaTime** [comment]
@taiseiiiii is it good now?

File: .changeset/real-vans-film.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (taiseiiiii) [inline .changeset/real-vans-film.md:1]: Please delete this file.

**Review by taiseiiiii — CHANGES_REQUESTED** [review]
@KekmaTime 
Please use one changeset per PR.

File: .changeset/real-vans-film.md (hunk 4 above)
Comment (KekmaTime) [reply .changeset/real-vans-film.md:1]: okay sorry about that

**Review by taiseiiiii — APPROVED** [review]
@KekmaTime 
I appreciate you addressing this

LGTM!!

**Comment by KekmaTime** [comment]
> @KekmaTime 
> I appreciate you addressing this
> 
> LGTM!!

👍

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**